components/slider.js

```diff
@@ -1,23 +1,63 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
-import animateScrollTo from 'animated-scroll-to';
+import scrollToWithAnimation from 'scrollto-with-animation';
 import { Highlight } from '../components/highlights'
 import stylesheet from './slider.scss'
 import { ArrowButton } from './arrow-button';
 import { MediaList } from './icon';
 import { IndustriesMain } from './industries-main';
 
+const autoSliderTime = 15000;
+
 export class Slider extends React.Component {
   constructor(props) {
     super(props);
     this.state = {currentScroll: 0, isBeginning: true, isEnd: false}
     this.scroll = this.scroll.bind(this);
+    this.scrollAutomatically = this.scrollAutomatically.bind(this);
     this.list = this.props.listRef;
+    this.shouldScrollRight = true;
 
     this.setListRef = element => {
       this.list = element;
     };
+
+    if(this.props.shouldScrollAutomatically){
+      this.scrollAutomatically();
+    }
+  }
+
+  scrollAutomatically(){
+    const element = this.list;
+    if(!element){
+      setTimeout(() => this.scrollAutomatically(), 500);
+      return;
+    }
```

Comment:
I'm not sure I follow this logic, but in general to me it seems that if we are returning early from this function from a condition, we can have that condition earlier. Let's avoid early returns since they usually mean the function is trying to do too much.

An alternative could be to check earlier in the call of `scrollAutomatically` to check `element`. 

```
const { list, shouldScrollAutomatically } = this.props;
shouldScrollAutomatically && list && this.scrollAutomatically()
```

Furthermore, you can split this functionality into two functions, something like `scrollAutomaticallyWithoutElement` and `scrollAutomatically`.

Reply:
This is related to the explanation about the reference below. I don't actually like it much either. But the reference is not defined right at the start when Slider actually mounts.